inst/tests/tests.Rraw

```diff
@@ -6031,6 +6031,19 @@ thisDT <- copy(DT)[, c("aaa", "b") := 2]
 test(1419.58, indices(thisDT), c("a", "ab"))
 test(1419.59, allIndicesValid(thisDT), TRUE)
 
+## setkey on same col as index before
+DT <- data.table(a = c(1,1,1,1,1,2,2,2,2,2),
+                 aaa = c(1,1,2,2,2,1,1,2,2,2))
+setindex(DT, a)
+test(1419.60, allIndicesValid(DT), TRUE)
+test(1419.61, setkey(DT, a, verbose=TRUE), output="using existing index for a")
+DT <- data.table(a = c(1,1,1,1,1,2,2,2,2,2),
+                 aaa = c(1,1,2,2,2,1,1,2,2,2),
+                 bbb = c(1,1,2,0,1,1,1,0,1,1))
+setindex(DT, a)
+setindex(DT, aaa)
+test(1419.62, allIndicesValid(DT), TRUE)
+test(1419.63, setkey(DT, aaa, verbose=TRUE), output="using existing index for aaa")
```

Comment:
Same request here as above for 1419.61.  This test needs to check the result of this setkey() command has done the correct thing (e.g. changed the physical row order properly) as well as doing that for the right reason (`output=`).
Since setkey() returns DT invisibly, the easiest way is just to pass y=data.table(...) into this test().  See other tests() in his file for examples.